app/code/controller/CustomTaxonomyDirectoryController.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\Controller;
+
+use Page;
+use SilverStripe\View\ArrayData;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\CMS\Model\SiteTree;
+use SilverStripe\GraphQL\Controller;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Taxonomy\TaxonomyTerm;
+use SilverStripe\Taxonomy\Controllers\TaxonomyDirectoryController;
+
+class CustomTaxonomyDirectoryController extends TaxonomyDirectoryController
+{
+    private static $allowed_actions = [
+        'showTags'
+    ];
+
+    /**
+     * This action allows for pages to be filtered by a desired term
+     *
+     * @param HTTPRequest $request
+     * @return void
+     */
+    public function showTags(HTTPRequest $request)
+    {
+        $termString = $request->param('ID');
+
+        // Used to handle empty ID requests
+        $title = TaxonomyTerm::get()->byID($termString) ? TaxonomyTerm::get()->byID($termString)->Name : '';
+
+        $pages = Page::get()->filter(['Terms.Name' => $title]);
```

Comment:
Theoretically this should still work, could you change `Terms.Name` to `Terms.ID` this way we can be more confident that it the value is unique, also update `$title` to `$termID`

Reply:
Might also be good to do a `turnary` check on `$termID`, similar to the `title` above